Add Bulk controller and unit tests (#21)
kuzzleio/sdk-csharp: doc/1/controllers/bulk/import-async/index.md

---
code: true
type: page
title: ImportAsync
desription: Creates, updates or deletes large amounts of documents as fast as possible.
---

# ImportAsync

Creates, updates or deletes large amounts of documents as fast as possible.

This route is faster than the document:m* routes family (e.g. document:mCreate), but no real-time notifications will be generated, even if some of the documents in the import match subscription filters.

If some documents actions fail, the client will receive a PartialError error.

## Signature

```csharp
async Task<JObject> ImportAsync(
    string index,
    string collection,
    JArray bulkData

> )

Reviewer: Should be in the upper line

```

## Arguments

| Arguments    | Type              | Description                                                  |
| ------------ | ----------------- | ------------------------------------------------------------ |
| `index`      | <pre>string</pre> | Index name                                                   |
| `collection` | <pre>string</pre> | Collection name                                              |
| `bulkData`   | <pre>JArray</pre> | Bulk operations to perform, following ElasticSearch Bulk API |

## Return

> An object containing information about the import status for each document.

Reviewer: ```suggestion
A `JObject` containing information about the import status for each document.
```
